File: backend/src/main/java/com/dongnemarket/auction/dto/AuctionStateResponse.java
package com.dongnemarket.auction.dto;

import com.dongnemarket.auction.entity.Auction;
import com.dongnemarket.auction.entity.AuctionStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public class AuctionStateResponse {

    private final Long auctionId;
    private final BigDecimal currentPrice;
    private final Long highestBidderId;
    private final AuctionStatus status;
    private final LocalDateTime endAt;

    private AuctionStateResponse(Long auctionId, BigDecimal currentPrice, Long highestBidderId,
                                 AuctionStatus status, LocalDateTime endAt) {
        this.auctionId = auctionId;
        this.currentPrice = currentPrice;
        this.highestBidderId = highestBidderId;
        this.status = status;
        this.endAt = endAt;
    }

    public static AuctionStateResponse from(Auction auction) {
        return new AuctionStateResponse(
                auction.getId(), auction.getCurrentPrice(), auction.getHighestBidderId(),
                auction.getStatus(), auction.getEndAt());
    }

    public Long getAuctionId() { return auctionId; }
    public BigDecimal getCurrentPrice() { return currentPrice; }
    public Long getHighestBidderId() { return highestBidderId; }
    public AuctionStatus getStatus() { return status; }
    public LocalDateTime getEndAt() { return endAt; }
}

File: backend/src/main/kotlin/com/dongnemarket/auction/dto/AuctionStateResponse.kt
package com.dongnemarket.auction.dto

import com.dongnemarket.auction.entity.Auction
import com.dongnemarket.auction.entity.AuctionStatus
import java.math.BigDecimal
import java.time.LocalDateTime

/** 입찰·종료 시 `/topic/auction/{id}` 로 브로드캐스트하는 경매 현재 상태(현재가·최고입찰자·상태). */
@ConsistentCopyVisibility
data class AuctionStateResponse private constructor(
    val auctionId: Long?,
    val currentPrice: BigDecimal,
    val highestBidderId: Long?,
    val status: AuctionStatus,
    val endAt: LocalDateTime,
) {
    companion object {
        @JvmStatic
        fun from(auction: Auction): AuctionStateResponse =
            AuctionStateResponse(
                auction.id,
                auction.currentPrice,
                auction.highestBidderId,
                auction.status,
                auction.endAt,
            )
    }
}
